odoo/fields.py

```diff
@@ -1861,32 +1865,51 @@ def _get_attrs(self, model_class, name):
     _description_strip_classes = property(attrgetter('strip_classes'))
 
     def convert_to_column(self, value, record, values=None, validate=True):
-        if value is None or value is False:
-            return None
-        if self.sanitize:
-            return html_sanitize(
-                value, silent=True,
-                sanitize_tags=self.sanitize_tags,
-                sanitize_attributes=self.sanitize_attributes,
-                sanitize_style=self.sanitize_style,
-                sanitize_form=self.sanitize_form,
-                strip_style=self.strip_style,
-                strip_classes=self.strip_classes)
-        return value
+        return self._convert(value, record, True)
 
     def convert_to_cache(self, value, record, validate=True):
+        return self._convert(value, record, validate)
+
+    def _convert(self, value, record, validate):
         if value is None or value is False:
             return None
-        if validate and self.sanitize:
-            return html_sanitize(
-                value, silent=True,
-                sanitize_tags=self.sanitize_tags,
-                sanitize_attributes=self.sanitize_attributes,
-                sanitize_style=self.sanitize_style,
-                sanitize_form=self.sanitize_form,
-                strip_style=self.strip_style,
-                strip_classes=self.strip_classes)
-        return value
+
+        if not validate or not self.sanitize:
+            return value
+
+        sanitize_vals = {
+            'silent': True,
+            'sanitize_tags': self.sanitize_tags,
+            'sanitize_attributes': self.sanitize_attributes,
+            'sanitize_style': self.sanitize_style,
+            'sanitize_form': self.sanitize_form,
+            'strip_style': self.strip_style,
+            'strip_classes': self.strip_classes
+        }
+
+        if self.sanitize_overridable and not record.env.registry._init:
```

Comment:
The `not record.env.registry._init` part is needed as we still want to `html_sanitize` the content during the XML import.
Indeed, some records' fields' values are ("wrongly"?) set as `type="xml"` while they are pure HTML value inside an HTML field. Without going through the sanitizer, the value will be prefixed/wrapped into

```xml
<?xml version="1.0"?>                                                                           +
<data><!-- ACTUAL VALUE --></data>
```
(wrapped only if multiple root nodes)

Thus, this PR would change the value added in DB for such cases.

It would then mean that for those (very rare) cases, if a restricted editor would try to modify it, it would directly be catch as unsafe modification due to the `<data/>` wrapper.

Note that I made a PR to "fix" those occurrences: https://github.com/odoo/odoo/pull/98239

-------------

Initial TMP fix was to also check for similarity with the `<data/>` wrapper, but:
1. It was doing some extra operation / slowing down the process for no reason
2. For those rare cases, it was inserting values in DB differently than before the PR (wrapping into `<data/>`) 

```py
    and get_parsed(original_value) != get_parsed('<data>%s</data>' % initial_value_sanitized)
```